pages/learn/deploy/deployment.md

```diff
@@ -1,52 +1,250 @@
+![how balena push works](/img/common/deployment/balena-push.jpg)
 
-The method our builders use to build your code depends on the project you are pushing. If your project includes a Dockerfile, the builders will execute every command in the Dockerfile, from including the base OS you define, to installing packages, pulling git repositories, and running any other commands you specify.
+Once the cloud builder has successfully completed building all the images in the deployment, it will upload these images to the balenaCloud registry and create a release on the API. It will then notify all the devices in the fleet that a new release is available. If you need to pull in proprietary code or use a private base image during your builds you can do so using the [build time secrets](#build-time-secrets-and-variables) or [private base images](#private-base-images) feature of `balena push`.
 
-Alternatively, if your project includes a package.json (i.e. a Node.js project), then that will be used instead to create an implicit Dockerfile, which simulates the build process a node.js/npm project expects. In this way, we are able to transparently run node.js projects on {{ $names.company.lower }}, while taking advantage of some of Docker’s caching features.
+It should be noted that `balena push` is independent of git, so you are free to use any version control system you wish. This also means that it is possible to use [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) in your project when deploying with `balena push`.
 
-A Dockerfile will always give you more power to fine-tune the build process, but it’s also good to start fast without one, since you can shift to a Dockerfile whenever you like. We will soon add more languages to the “first-class” support we give to Node.js. Until then, any other language can be used by adding a Dockerfile. It’s important to note that the environment within which your code gets built will match the devices you use for your app. So if you’re for example pushing to an app containing BeagleBone Black devices, we’ll build your code in an ARMv7 environment, while for the Raspberry Pi B+ it would be ARMv6.
+### Additional Options
+#### `--source, -s <source>`                    
 
-If you push a project with only a `Dockerfile`, `Dockerfile.template`, or `package.json`, a single container image will be built and sent to your device. The single container will show up on the device dashboard as a service with the name `main`. 
+The `--source` flag allows you do define a path to your source code folder. This flag directs `push` to send that directory to be built on either the [cloud builder](#the-balenacloud-build-server) of [local device](/learn/develop/local-mode/). You should ensure your folder follows the [standard balena project structure](#project-structure).
```

Comment:
should that "of" be an "or"? So it reads "either the cloud builder *or* the local device"?